test/e2e/framework.go

```diff
@@ -1481,6 +1490,18 @@ func (b *PodBuilder) WithReadinessProbe(probe *corev1.Probe) *PodBuilder {
 	return b
 }
 
+// WithCustomDNSConfig adds custom DNS IP to the Pod spec.
+func (b *PodBuilder) WithCustomDNSConfig(dnsServiceIP string) *PodBuilder {
```

Comment:
I thought you would make it `WithCustomDNSConfig(dnsConfig *corev1.PodDNSConfig)`. **I think your approach is fine**, but you should rename the function to `WithCustomDNSServer` for accuracy.

Reply:
Thank you for mentioning this, but i feel the idea you had earlier about passing a ```*corev1.PodDNSConfig``` instance would be more appropriate and be clean.